src/main/scala/codecheck/github/operations/WebhookOp.scala

```diff
@@ -0,0 +1,81 @@
+package codecheck.github.operations
+
+import scala.concurrent.Future
+import scala.concurrent.ExecutionContext.Implicits.global
+import org.json4s.JArray
+
+import codecheck.github.api.GitHubAPI
+import codecheck.github.exceptions.NotFoundException
+import codecheck.github.models.Webhook
+import codecheck.github.models.WebhookConfig
+import codecheck.github.models.WebhookUpdateInput
+import codecheck.github.models.WebhookInput
+
+trait WebhookOp {
+  self: GitHubAPI =>
+
+  def listWebhooks(owner: String, repo: String): Future[List[Webhook]] = {
+    self.exec("GET", s"/repos/${owner}/${repo}/hooks").map { 
+      _.body match {
+        case JArray(arr) => arr.map(new Webhook(_))
+        case _ => throw new IllegalStateException()
+      }
+    }
+  }
+
+  def getWebhook(owner: String, repo: String, id: Long): Future[Option[Webhook]] = {
+  	self.exec("GET", s"/repos/${owner}/${repo}/hooks/${id}").map { res =>
+  		res.statusCode match {
+  			case 404 => None
+  			case 200 => Some(new Webhook(res.body))
+  		}
+  	}
+  } 
+
+  def createWebhook(owner: String, repo: String, input: WebhookInput): Future[Option[Webhook]] = {
+  	self.exec("POST", s"/repos/${owner}/${repo}/hooks", input.value, false).map { res =>
+  		res.statusCode match {
+  			case 404 => None
+  			case 201 => Some(new Webhook(res.body))
+  		}
+  	}
+  }
+
+  //It is apparently an issue with Github's Webhook API that add_events and remove_events cannot be done
+  //in a single operation. To add and remove events, it must be done through two seperate calls of updateWebhook.
+  def updateWebhook(owner: String, repo: String, id: Long, input: WebhookUpdateInput): Future[Option[Webhook]] = {
```

Comment:
If using invalid organization or repo, it will 404. Normally it returns the updated Webhook, but when 404 it should return None.